src/js/game/hud/parts/mass_selector.js

```diff
@@ -243,9 +244,18 @@ export class HUDMassSelector extends BaseHUDPart {
 
             for (let x = realTileStart.x; x <= realTileEnd.x; ++x) {
                 for (let y = realTileStart.y; y <= realTileEnd.y; ++y) {
-                    const contents = this.root.map.getLayerContentXY(x, y, this.root.currentLayer);
-                    if (contents && this.root.logic.canDeleteBuilding(contents)) {
-                        this.selectedUids.add(contents.uid);
+                    let entities = [];
+                    if (
+                        this.root.keyMapper.getBinding(KEYMAPPINGS.massSelect.massSelectSelectMultiLayer)
+                            .pressed
+                    )
+                        entities = this.root.map.getLayersContentsMultipleXY(x, y);
+                    else entities = [this.root.map.getLayerContentXY(x, y, this.root.currentLayer)];

```

Comment:
I always use braces (`{}`), dunno why the linter didn't complain, but I can change that myself :)

Reply:
will change